actors/actor.go

```diff
@@ -0,0 +1,48 @@
+/*interface file for actors*/
+package actors
+
+import (
+	"fmt"
+	"student20_pop/db"
+	"student20_pop/define"
+)
+
+type Actor interface {
+	//Public functions
+	HandleWholeMessage(msg []byte, userId int) ([]byte, []byte, []byte)
+	//Private functions
+	handlePublish(generic define.Generic) ([]byte, []byte, error)
+	handleCreateLAO(message define.Message, channel string, generic define.Generic) ([]byte, []byte, error)
+	handleUpdateProperties(message define.Message, channel string, generic define.Generic) ([]byte, []byte, error)
+	handleWitnessMessage(message define.Message, channel string, generic define.Generic) ([]byte, []byte, error)
+	handleLAOState(message define.Message, channel string, generic define.Generic) ([]byte, []byte, error)
+	handleCreateRollCall(message define.Message, channel string, generic define.Generic) ([]byte, []byte, error)
+}
+
+//general actors functions, act only in the "Sub" database
+func handleSubscribe(generic define.Generic, userId int) error {
```

Comment:
The name generic came from the protocol (https://github.com/dedis/student20_pop/blob/proto-specs/genericMessage.json), however it was not up to date anymore, so we rename it to query (to match the nomenclature https://github.com/dedis/student20_pop/blob/proto-specs/query/query.json), and added one more level above, called genericMessage which englobes now both queries and answers

Reply:
Kudos for matching up the code with the spec. That's an excellent decision.
And you're right that this would be perfectly clear to someone reading the code, if the package name reflected that.
Say it were called `jsonrpc.Generic`, I probably wouldn't even have asked you the question, but `define` makes it very hard to guess :)